File: apps/api/app/schemas/health.py
"""Health and service information schemas."""

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: str
    service: str
    version: str
    environment: str
    dependencies: "DependenciesResponse"


class DatabaseDependencyResponse(BaseModel):
    status: str
    postgis: bool | None


class DependenciesResponse(BaseModel):
    database: DatabaseDependencyResponse
    weather_provider: "ProviderDependencyResponse"
    marine_provider: "ProviderDependencyResponse"


class ProviderDependencyResponse(BaseModel):
    status: str
    last_success: str | None = None


class APIInfoResponse(BaseModel):
    name: str
    status: str
    docs: str

File: src/lib/api/health.ts
import { apiClient } from "@/lib/api/client";
import { API_V1_PREFIX } from "@/lib/api/config";
export interface HealthResponse { status: "healthy" | "degraded"; service: "orca-api"; version: string; environment: string; dependencies: { database: { status: "healthy" | "unavailable"; postgis: boolean | null } }; }
export const getHealth = (): Promise<HealthResponse> => apiClient<HealthResponse>(`${API_V1_PREFIX}/health`);
